File: test_server.py
#!/usr/bin/env python3

import requests
import json
import time
import subprocess
import signal
import sys
import os

def test_fido_endpoints():
    # Basic test payload for attestation options
    attestation_request = {
        "username": "test@example.com",
        "displayName": "Test User",
        "authenticatorSelection": {
            "requireResidentKey": False,
            "authenticatorAttachment": "cross-platform",
            "userVerification": "preferred"
        },
        "attestation": "direct"
    }
    
    print("Testing /attestation/options endpoint...")
    
    try:
        response = requests.post(
            "http://localhost:8080/attestation/options",
            json=attestation_request,
            headers={"Content-Type": "application/json"},
            timeout=10
        )
        print(f"Status Code: {response.status_code}")
        print(f"Response Headers: {response.headers}")
        print(f"Response Body: {response.text}")
        
        if response.status_code == 200:
            print("✅ /attestation/options working!")
            data = response.json()
            print(f"Challenge: {data.get('challenge', 'N/A')}")
        else:
            print("❌ /attestation/options failed")
            
    except requests.exceptions.RequestException as e:
        print(f"❌ Request failed: {e}")
    except Exception as e:
        print(f"❌ Unexpected error: {e}")

    # Test health check
    try:
        print("\nTesting /health endpoint...")
        response = requests.get("http://localhost:8080/health", timeout=5)
        print(f"Health Status: {response.status_code}")
        print(f"Health Response: {response.text}")
    except Exception as e:
        print(f"Health check failed: {e}")

if __name__ == "__main__":
    test_fido_endpoints()

File: test_server.rs
//! Simple test server for FIDO endpoints

use actix_cors::Cors;
use actix_web::{middleware::Logger, web, App, HttpServer, HttpResponse};
use serde_json::json;

// Simple health check endpoint
async fn health_check() -> actix_web::Result<HttpResponse> {
    Ok(HttpResponse::Ok().json(json!({
        "status": "ok",
        "service": "FIDO Server Test",
        "version": "0.1.0"
    })))
}

// Mock attestation options endpoint
async fn attestation_options(req: web::Json<serde_json::Value>) -> actix_web::Result<HttpResponse> {
    println!("Received attestation options request: {}", serde_json::to_string_pretty(&req.into_inner()).unwrap());
    
    let response = json!({
        "status": "ok",
        "errorMessage": "",
        "rp": {
            "name": "Example Corporation"
        },
        "user": {
            "id": "S3932ee31vKEC0JtJMIQ",
            "name": "johndoe@example.com",
            "displayName": "John Doe"
        },
        "challenge": "uhUjPNlZfvn7onwuhNdsLPkkE5Fv-lUN",
        "pubKeyCredParams": [
            {
                "type": "public-key",
                "alg": -7
            }
        ],
        "timeout": 10000,
        "excludeCredentials": [],
        "authenticatorSelection": {
            "requireResidentKey": false,
            "authenticatorAttachment": "cross-platform",
            "userVerification": "preferred"
        },
        "attestation": "direct"
    });
    
    Ok(HttpResponse::Ok().json(response))
}

// Mock attestation result endpoint
async fn attestation_result(req: web::Json<serde_json::Value>) -> actix_web::Result<HttpResponse> {
    println!("Received attestation result request: {}", serde_json::to_string_pretty(&req.into_inner()).unwrap());
    
    let response = json!({
        "status": "ok",
        "errorMessage": ""
    });
    
    Ok(HttpResponse::Ok().json(response))
}

// Mock assertion options endpoint
async fn assertion_options(req: web::Json<serde_json::Value>) -> actix_web::Result<HttpResponse> {
    println!("Received assertion options request: {}", serde_json::to_string_pretty(&req.into_inner()).unwrap());
    
    let response = json!({
        "status": "ok",
        "errorMessage": "",
        "challenge": "6283u0svT-YIF3pSolzkQHStwkJCaLKx",
        "timeout": 20000,
        "rpId": "localhost",
        "allowCredentials": [
            {
                "id": "m7xl_TkTcCe0WcXI2M-4ro9vJAuwcj4m",
                "type": "public-key"
            }
        ],
        "userVerification": "required"
    });
    
    Ok(HttpResponse::Ok().json(response))
}

// Mock assertion result endpoint
async fn assertion_result(req: web::Json<serde_json::Value>) -> actix_web::Result<HttpResponse> {
    println!("Received assertion result request: {}", serde_json::to_string_pretty(&req.into_inner()).unwrap());
    
    let response = json!({
        "status": "ok",
        "errorMessage": ""
    });
    
    Ok(HttpResponse::Ok().json(response))
}

#[actix_web::main]
async fn main() -> std::io::Result<()> {
    env_logger::init_from_env(env_logger::Env::new().default_filter_or("info"));

    println!("Starting FIDO Test Server on http://localhost:8080");

    HttpServer::new(|| {
        let cors = Cors::default()
            .allow_any_origin()
            .allow_any_method()
            .allow_any_header()
            .max_age(3600);

        App::new()
            .wrap(Logger::default())
            .wrap(cors)
            .route("/health", web::get().to(health_check))
            .route("/attestation/options", web::post().to(attestation_options))
            .route("/attestation/result", web::post().to(attestation_result))
            .route("/assertion/options", web::post().to(assertion_options))
            .route("/assertion/result", web::post().to(assertion_result))
    })
    .bind("127.0.0.1:8080")?
    .run()
    .await
}